liquidity_pool/src/liquidity.rs

```diff
@@ -335,14 +367,29 @@ pub trait LiquidityModule:
 
         self.borrow_position(borrow_position_nonce).clear();
 
+        let bp = self.get_base_precision();
+        let lend_amount_to_return = (asset_amount * (&bp + &liquidation_bonus)) / bp;
+        let lend_tokens = borrow_position.lend_tokens.clone();
```

Comment:
Use reference instead of cloning
```suggestion
        let lend_tokens = &borrow_position.lend_tokens;
```